translate/src/util.rs

```diff
@@ -0,0 +1,107 @@
+//! Utility functions that don't really belong in any other module.
+
+use std::fs::{create_dir, metadata, read_dir, remove_dir_all, remove_file};
+use std::io::{self, ErrorKind::NotFound};
+use std::path::Path;
+use thiserror::Error;
+
+/// Sets up an empty, writable directory at the given path. Used to create the output directory and
+/// diagnostics directory (if specified on the command line). If the directory:
+/// 1. Does not exist: will create the directory.
+/// 2. Exists and is empty: will verify the directory is writable.
+/// 3. Exists and is nonempty: if `delete_contents` is true, the contents will be deleted. If
+///    `delete_contents` is false, will return `Err(EmptyDirError::NonEmpty)`.
+pub fn empty_writable_dir<P: AsRef<Path>>(
+    path: P,
+    delete_contents: bool,
+) -> Result<(), EmptyDirError> {
+    let entries = match read_dir(&path) {
```

Comment:
I believe this will also fail if the path is _not_ a directory (but a file). Which we want, so great.

Reply:
Test added; yes that errored as desired.